.github/workflows/prettier.yml

```diff
@@ -0,0 +1,39 @@
+name: Prettier
+
+# This action works with pull requests and pushes
+on:
+  - push
+
+jobs:
+  prettier:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          # Make sure the actual branch is checked out when running on pull requests
+          ref: ${{ github.head_ref }}
+
+      # Make sure @sourcegraph/prettierrc is installed
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+          cache: yarn
+        run: yarn
+
+      - name: Read package.json
+        id: read_package_json
+        run: |
+          JSON=$(cat package.json)
+          echo "::set-output name=packageJson::${JSON//'%'/'%25'}"
+
+      - name: Prettier
```

Comment:
Awesome automation! What do you think about running prettier only against changed files on CI?
It seems that such an option is available [out of the box](https://github.com/creyD/prettier_action#example-2-using-the-only_changed-or-same_commit-option-on-pr).

Reply:
I honestly don't trust it enough for the benefit it would provide (performance). It is implemented with `git diff --name-only --diff-filter=d HEAD^..HEAD`, which already I am not sure if it will handle merge commits correctly, which appear frequently when people click "Update branch" in GitHub. Prettier is already pretty fast (haha), probably the fastest build step from the ones we have, so I don't think it's worth the risk/complexity over just formatting all files on every run.